plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -555,6 +562,10 @@ public void dropDatabase(HiveIdentity identity, String databaseName, boolean del
         catch (AmazonServiceException e) {
             throw new TrinoException(HIVE_METASTORE_ERROR, e);
         }
+
+        if (deleteData) {
+            location.ifPresent(path -> deleteDir(hdfsContext, hdfsEnvironment, new Path(path), true));
```

Comment:
it's an error when `deleteData && location.isEmpty()`.
Either `checkState` or add  a comment why we're not failing

Reply:
This can theoretically occur if `database.getLocation()` returns empty.

In this case, it only occurs if Glue's API returns null for the database location, which seems unlikely, but that method is not documented as non-null.